addons/stock/static/src/js/report_stock_reception.js

```diff
@@ -0,0 +1,214 @@
+/** @odoo-module alias=stock.ReceptionReport **/
+
+import clientAction from 'report.client_action';
+import core from 'web.core';
+
+const qweb = core.qweb;
+
+const ReceptionReport = clientAction.extend({
+    /**
+     * @override
+     */
+    init: function (parent, action, options) {
+        this._super.apply(this, arguments);
+        this.context = Object.assign(action.context || {}, {
+            active_ids: action.context.default_picking_ids,
+        });
+        this.report_name = `stock.report_reception`
+        this.report_url = `/report/html/${this.report_name}/?context=${JSON.stringify(this.context)}`;
+        this._title = action.name;
+    },
+
+    /**
+     * @override
+     */
+     start: function () {
+        return Promise.all([
+            this._super(...arguments),
+        ]).then(() => {
+            this._renderButtons();
+        });
+    },
+
+    /**
+     * @override
+     */
+    on_attach_callback: function () {
+        this._super();
+        this.iframe.addEventListener("load",
+            () => this._bindAdditionalActionHandlers(),
+            { once: true }
+        );
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Renders extra report buttons in control panel
+     */
+     _renderButtons: function () {
+        this.$buttons.append(qweb.render('reception_report_buttons', {}));
+        this.$buttons.on('click', '.o_report_reception_assign', this._onClickAssign.bind(this));
+        this.$buttons.on('click', '.o_print_label', this._onClickPrintLabel.bind(this));
+        this.controlPanelProps.cp_content = {
+            $buttons: this.$buttons,
+        };
+    },
+
+    /**
+     * Bind additional <button> action handlers
+     */
+    _bindAdditionalActionHandlers: function () {
+        let rr = $(this.iframe).contents().find('.o_report_reception');
+        rr.on('click', '.o_report_reception_assign', this._onClickAssign.bind(this));
+        rr.on('click', '.o_report_reception_unassign', this._onClickUnassign.bind(this));
+        rr.on('click', '.o_report_reception_forecasted', this._onClickForecastReport.bind(this));
+        rr.on('click', '.o_print_label', this._onClickPrintLabel.bind(this));
+    },
+
+
+    _switchButton: function(button) {
```

Comment:
Always a space before params (except if it's a named fonction, cf. the [JS coding guideline](https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines))
```suggestion
    _switchButton: function (button) {
```
Same comment for following methods